packages/frontend/@n8n/design-system/src/components/N8nPromptInput/N8nPromptInput.vue

```diff
@@ -61,13 +71,64 @@ const { characterCount, isOverLimit, isAtLimit } = useCharacterLimit({
 
 const showWarningBanner = computed(() => isAtLimit.value);
 const sendDisabled = computed(
-	() => !textValue.value.trim() || props.streaming || props.disabled || isOverLimit.value,
+	() =>
+		!textValue.value.trim() ||
+		props.streaming ||
+		props.disabled ||
+		isOverLimit.value ||
+		props.creditsRemaining === 0,
 );
 
 const containerStyle = computed(() => {
 	return { minHeight: isMultiline.value ? '80px' : '40px' };
 });
 
+const showCredits = computed(() => {
+	return (
+		props.creditsQuota !== undefined &&
+		props.creditsRemaining !== undefined &&
+		props.creditsQuota !== -1
```

Comment:
nitpick: could we make this a named constant to describe the scenario in which `-1` occurs?